tests/core/client.spec.ts

```diff
@@ -0,0 +1,86 @@
+import * as elliptic from 'elliptic';
+import { BTCPayClient } from '../../src/core/client';
+import { Cryptography as myCrypto } from '../../src/core/cryptography';
+
+const USER_NAME = 'test@example.com';
+const PASSWORD = 'satoshinakamoto';
+const URL = 'https://testnet.demo.btcpayserver.org/';
+
+const MY_PRIVATE_KEY = Buffer.from(
+  '31eb31ecf1a640c9d1e0a1105501f36235f8c7d51d67dcf74ccc968d74cb6b25',
+  'hex',
+);
+
+const STORE_ID = 'HPPHFtqtsKsF3KU18fBNwVGP64hicGoRynvQrC3R2Rkw';
+const TOKENS = {
+  merchant: 'DwSMQ4SF7GAJRaMiLn4zjAR35bFJwgSpuKt9pxYoQNjJ',
+};
+
+const INVOICE_ID = 'TRnwXeAkuLQihe22mJs7J4';
+
+// We need a way to programmatically get a new pairing code...
+const SERVER_PAIRING_CODE = 'apYAxP9';
+
+let MY_KEYPAIR: elliptic.ec.KeyPair;
+let client: BTCPayClient;
+describe('btcpay.core.cryptography', () => {
+  beforeAll(() => {
+    MY_KEYPAIR = myCrypto.load_keypair(MY_PRIVATE_KEY);
+    client = new BTCPayClient(URL, MY_KEYPAIR, TOKENS);
+  });
+
+  it('should pair with server', async () => {
+    const myClient = new BTCPayClient(URL, MY_KEYPAIR);
+    const result = await myClient.pair_client(SERVER_PAIRING_CODE).then(
```

Comment:
whou you don't use try/catch?

Reply:
because "if specific error return x" with try catch makes long try catch indent hell.

Or I have to use let.

const > let

Reply:
it is only one... so maybe it's ok, but just a habit.